src/token.jl

```diff
@@ -95,6 +95,204 @@ function kind(t::AbstractToken)
     return t.kind
 end
 exactkind(t::AbstractToken) = t.kind
+
+Meta.parse(t::T) where {T <: Union{Token, Array{Token}}} = Meta.parse(untokenize(t))
+
+"""
+    UndefToken
+
+An abstract type that shows if the an evaluated `Token` is defined.
+"""
+abstract type UndefToken end
+
+# Note: in the following, Symbol and Expr methods are meant for seemless and errorless experience.
+
+"""
+    tisdefined(t)
+    tisdefined(modul, t)
+
+Check if a `Token` that represent a `Symbol` is defined. Throws error if the `Token` is representing an `Expr`.
+
+# Examples
+```julia
+julia> t = collect(tokenize("Int64"))
+
+julia> tisdefined(t)
+true
+```
+"""
+function tisdefined(modul, t::T) where {T <: Union{Token, Array{Token}, String}}
+    pt = Meta.parse(t)
+    return isdefined(modul,pt)
+end
+
+tisdefined(t::T) where {T <: Union{Token, Array{Token}, String}} = tisdefined(@__MODULE__, t)
```

Comment:
This `@__MODULE__` is equivalent to writing `Tokenize`. It will not expand to the module of the caller. For that you need to use a macro and `__module__`.